pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,378 @@
+        if len(shown_charts) != 0:
+            top_level = shown_charts[list(shown_charts)[-1]]
+            # FROM "https://github.com/pyqtgraph/pyqtgraph/pull/2010" by herodotus77
+            # propagate mouse actions to charts "hidden" behind
+            for k, c in self.charts.items():
+                self.connect_signals(top_level, c)
+        # MOVE LEGEND TO LAYOUT
+        if self.pi.legend is not None:
+            self.pi.legend.setParentItem(self.pi)
+        self.update()
+
+    def show_axes(self, axes=None):
+        """Shows all the selected axes."""
+        # SELECT AND ASSEMBLE AXES
+        if axes is None:
+            axes = list(self.axes)
+        lo = {
+            "left": [],
+            "right": [],
+            "top": [],
+            "bottom": [],
+        }
+        for k, a in self.axes.items():
+            if k in axes:
+                lo[a.orientation].append(a)
+        vx = len(lo["left"])
+        vy = 1 + len(lo["top"])
+        # ADD TITLE ON TOP
+        self.pi.titleLabel.show()
+        self.layout.addItem(self.pi.titleLabel, 0, vx)
+        # ADD MAIN PLOTITEM
+        self.vb.show()
+        self.layout.addItem(self.vb, vy, vx)
+        # ADD AXIS
+        shown_axes = {"x": {}, "y": {}}
+        for x, a in enumerate(lo["left"] + [None] + lo["right"]):
+            if a is not None:
+                a.show()
+                self.layout.addItem(a, vy, x)
+                shown_axes["x"][a.name] = a
+        for y, a in enumerate([None] + lo["top"] + [None] + lo["bottom"]):
+            if a is not None:
+                a.show()
+                self.layout.addItem(a, y, vx)
+                shown_axes["y"][a.name] = a
+        return shown_axes
+
+    def show_charts(self, charts=None):
+        """Shows all the selected charts."""
+        # SELECT CHARTS
+        if charts is None:
+            charts = self.charts
+        shown_charts = {}
+        for k, c in self.charts.items():
+            if k in charts:
+                c.show()
+                shown_charts[k] = c
+            else:
+                c.hide()
+        return shown_charts
+
+    def connect_signals(self, top_level, chart):
+        """Connects all signals related to this widget for the given chart given the top level one."""
+        self.disconnect_all(chart)
+        tvb = top_level.plotItem.vb
+        cvb = chart.plotItem.vb
+        scene = self.scene()
+        for axis_name in chart.axes:
+            # link axis to view
+            axis = self.axes[axis_name]
+            # connect view changes to axis
+            # set axis main view link if not assigned
+            if axis.linkedView() is None:
+                axis._linkedView = weakref.ref(cvb)
+            # FROM AxisItem.linkToView
+            # connect view changes to axis changes
+            if axis.orientation in ["right", "left"]:
+                chart.mapw_conn["cvb.sigYRangeChanged"] = cvb.sigYRangeChanged.connect(axis.linkedViewChanged)
```

Comment:
i can't since there are multiple signals connected with the same name from different objects, this is why i added that prefix. if you mean to use the actual signal object, i don't know if it's actually hashable and wouldn't collide with another with the same name

Reply:
could the dictionary key be `tuple(signal, slot)`, and the value by the `QMetaObject` that is returned when the `connect` method is called?

Reply:
i can do this but is it actually necessary? there are lambdas involved so it won't be a clean solution, also i don't see the use of this and would make the code less understandable in my opinion.
can i suggest using a string that describes the signal-connection use instead?

Reply:
some slots will introduce references to self inside self and would prevent garbage collection the same way the self in lambda does used a hacky workaround where needed.